django_app/redbox_app/redbox_core/views.py

```diff
@@ -82,25 +82,31 @@ def upload_view(request):
 
     if request.method == "POST":
         # https://django-storages.readthedocs.io/en/1.13.2/backends/amazon-S3.html
-        try:
-            uploaded_file: UploadedFile = request.FILES["uploadDoc"]
+        
+        uploaded_files: list[UploadedFile] = request.FILES.getlist("uploadDoc")
 
-            file_extension = Path(uploaded_file.name).suffix
+        if len(uploaded_files) == 0:
+            errors.append("No document selected")
 
+        for uploaded_file in uploaded_files:
+            file_extension = Path(uploaded_file.name).suffix
+            
             if uploaded_file.name is None:
                 errors.append("File has no name")
             if uploaded_file.content_type is None:
-                errors.append("File has no content-type")
+                errors.append(f"Error with {uploaded_file.name}: File has no content-type")
             if uploaded_file.size > MAX_FILE_SIZE:
-                errors.append("File is larger than 200MB")
+                errors.append(f"Error with {uploaded_file.name}: File is larger than 200MB")
             if file_extension not in APPROVED_FILE_EXTENSIONS:
-                errors.append(f"File type {file_extension} not supported")
-        except MultiValueDictKeyError:
-            errors.append("No document selected")
+                errors.append(f"Error with {uploaded_file.name}: File type {file_extension} not supported")
 
         if not errors:
-            errors += ingest_file(uploaded_file, request.user)
-
+            for uploaded_file in uploaded_files:
+                errors += ingest_file(uploaded_file, request.user) 
+        
+        # TO DO: Do we want to redirect anyway, as otherwise users won't see the files that have been ingested
```

Comment:
Yeah - we should discuss with Charlotte & maybe Rachael.

Reply:
I've had a catchup with Charlotte. I've got a plan - will provide an update in due course.